### PR TITLE
[BOT] chore: add new modal to trading view chart

### PR DESCRIPTION
**Pull request by shafin-deriv**
## Changes:

- Removed Jquery Modal for Trading View
- Added the new React-Based Draggable modal for Trading View
- Updated Draggable modal resize and drag logic
- Added dynamic z-index value to currently focused draggable modal

### CODE REVIEW

File: src/components/Draggable/draggable.scss [hunk 1]
```diff
@@ -19,13 +18,17 @@
             color: #fff;
             border: 1px solid var(--color-blue);
             border-radius: 3px;
+            width: 100%;
```
Comment (maryia-matskevich-deriv) [inline src/components/Draggable/draggable.scss:21]: in what case does it improve the layout for us?
Screenshot 2024-02-28 at 16 01 08

File: src/components/Draggable/Draggable.jsx [hunk 2]
```diff
@@ -45,38 +48,59 @@ const Draggable = ({
...
+        if (!draggableModals.length) return;
+        const maxZIndex = Array.from(draggableModals).reduce(
+            (maxZ, modal) => Math.max(maxZ, parseInt(window.getComputedStyle(modal).zIndex) || 0),
+            0
+        );
+        const newZIndex = maxZIndex + 3;
+        setZIndex(newZIndex);
+    };
+
     const handleMouseDown = (event, action) => {
```
Comment (maryia-matskevich-deriv) [inline src/components/Draggable/Draggable.jsx:65]: The handleMouseDown() function's length of **146 lines**😳🤯 already signals the violation of several principles of good code:
1. Single Responsibility Principle
2. Readability
3. Support and Testing
@shafin-deriv bhai, can we improve it somehow? 😌

Comment (maryia-matskevich-deriv) [reply src/components/Draggable/Draggable.jsx:65]: Screenshot 2024-02-28 at 16 23 50

**Comment by maryia-matskevich-deriv** [comment]
@shafin-deriv bhai, can make it without overlapping a modal window by the footer?

Screenshot 2024-02-28 at 15 52 24

File: src/components/Draggable/draggable.scss (hunk 1 above)
Comment (shafin-deriv) [reply src/components/Draggable/draggable.scss:21]: @maryia-matskevich-deriv as this component is going to be a shared draggable component so for the header I kept the width to 100% for any future case where it could be necessary. As the header is customizable by passing children as elements so I just kept it for any future use cases.

File: src/components/Draggable/draggable.scss [hunk 3]
```diff
@@ -19,13 +18,17 @@
...
             &__title {
                 font-size: 16px;
                 font-weight: 700;
             }
             &__close {
                 cursor: pointer;
             }
+            &-title {
+                width: 100%;
+            }
```
[the remark this answers is not in the document]
Comment (shafin-deriv) [reply src/components/Draggable/draggable.scss:31]: @maryia-matskevich-deriv same comment as the above one and if you check the log table modal then you will see it is breaking.
Screenshot 2024-02-29 at 11 07 48 AM

File: src/components/Draggable/Draggable.jsx (hunk 2 above)
Comment (shafin-deriv) [reply src/components/Draggable/Draggable.jsx:65]: @maryia-matskevich-deriv these 146 lines of code are actually doing a single task which is handling the mouse-down event. Now if you open the function itself you will see there are 4 other small functions within this function which have their own set of tasks. Now if I consider the principle of single responsibility I wouldn't think of it in terms of number of lines of code but rather how it is structured and whether it is hard to follow. In this case even if you check the react-rnd code they also have the resize functionality which is more than 100 lines of code but it was necessary to keep them together. In our case we can definitely separate all these functions but the calculations are dependant upon the closure of some of the variables on mouse down. Now we can do 2 things, keep them in separate functions outside of this function and pass all the necessary variables as parameters or keep them in separate functions but within the parent function scope so that all the shared variable values can be accessed within the closure. I chose the later one as it seems more understandable but I am open to suggestion as well.

Comment (maryia-matskevich-deriv) [reply src/components/Draggable/Draggable.jsx:65]: @shafin-deriv I agree with the statement about organizing variables within the parent function scope so that all shared variable values can be accessed within the closure. However, I would like to find evidence to support this hypothesis. In JavaScript, a new object is created for each function scope, which holds the necessary variables from the closure until they are no longer needed and are removed by the garbage collector.

Writing large functions can make the code more difficult to maintain and harder to read. Therefore, it would be beneficial to determine if the approach you have chosen is truly better, or if breaking it down into separate functions and organizing them in a separate "utils" file would result in the same efficiency in terms of JavaScript code computation.

I am genuinely interested in understanding this concept and finding evidence to support it. It is evident that libraries can be authored by developers like us and may not be perfect.

You are welcome to keep the code as it is, but I would greatly appreciate it if you could share any authoritative articles or evidence that you come across on this topic.

**Review by sandeep-deriv — APPROVED** [review]

**Review by rupato-deriv — APPROVED** [review]
